### ISSUE TITLE
Exemplify flag with multiple values

### ISSUE THREAD

**Issue by lorenzleutgeb** [oclif/oclif#184]
I want to request neither a feature nor report a bug. Instead, I want to request improved documentation of an existing feature.

From oclif output: `oclif/1.12.5 linux-x64 node-v10.11.0`. I am running Fedora 28. I am using TypeScript.

What I would like to have is oclif parsing the invocation `mycli mycommand -x a -x b` into the two values `'a'` and `'b'` for the flag `x` represented as an array.

My problem is that I do not know how to specify this flag in my command. I found `flags.string` and `flags.boolean` but neither matches my scenario. I tried spelling out the definition myself, but your typings are very hard to follow (for me, maybe not in general), and I got frustrated.

I ended up with something like:
```ts
  static flags = {
    schemas: {
      char: 's' as AlphabetLowercase,
      default: guessSchemas(),
      description: 'xxx',
      input: [],
      multiple: true,
      name: 'schemas',
      parse: (input: string, context: any) => [input], // This of course will not work as expected, but should be a valid definition.
      type: 'option',
    },
  }
```
which does not work:
```
error TS2345: Argument of type 'typeof GenerateGraphql' is not assignable to parameter of type 'Input<{ schemas: string[]; }>'.
  Types of property 'flags' are incompatible.
    Type '{ schemas: { char: AlphabetLowercase; default: string[]; description: string; input: never[]; multiple: boolean; name: string; parse: (input: string, context: any) => string[]; type: string; }; }' is not assignable to type 'Input<{ schemas: string[]; }>'.
      Types of property 'schemas' are incompatible.
        Type '{ char: AlphabetLowercase; default: string[]; description: string; input: never[]; multiple: boolean; name: string; parse: (input: string, context: any) => string[]; type: string; }' is not assignable to type 'IFlag<string[]>'.
          Type '{ char: AlphabetLowercase; default: string[]; description: string; input: never[]; multiple: boolean; name: string; parse: (input: string, context: any) => string[]; type: string; }' is not assignable to type 'IOptionFlag<string[]>'.
            Type '{ char: AlphabetLowercase; default: string[]; description: string; input: never[]; multiple: boolean; name: string; parse: (input: string, context: any) => string[]; type: string; }' is not assignable to type '{ type: "option"; helpValue?: string | undefined; default?: string[] | ((context: DefaultContext<string[]>) => string[] | undefined) | undefined; multiple: boolean; input: string[]; options?: string[] | undefined; }'.
              Types of property 'type' are incompatible.
                Type 'string' is not assignable to type '"option"'.
```

I also searched GitHub (especially the Heroku CLI) for a working declaration but could not find one that matches my use case.

Therefore I request that you (a) provide an example on how to achieve this, (b) integrate this into example to be part of one of the two examplary CLIs.

**Comment by ytilis** [comment]
So I think you might just be a little confused about how to properly configure a normal flag. `flags.string` and `flags.boolean` represent the types of the value you're actually passing (in your example, the types of `a` and `b`), not the structure you should be expecting. So if I'm reading your example correctly, for `mycli mycommand -x a -x b`, you should be able to simply do:

```
  myFlagName: flags.string({
    char: 'x',
    multiple: true,
  }),
```

And when you call `flags.myFlagName`, it will have the value `['a', 'b']`. 

Basically the `multiple: true` attribute in the flag config automatically builds an array from the values for you. Does that answer your question? Or did I misunderstand what you're trying to achieve?
